### ISSUE TITLE
Debugee program arguments are not passed along to MemCheck/Valgrind.

### ISSUE THREAD

**Issue by Nathilion** [eranif/codelite#1660]
I'm currently working on set of programs that must be started with a specific string that acts as an identifier. When I run the regular debugger the string I typed in 'Project Settings -> General -> Program Arguments' is properly passed on to the program. But, when I try to start the program via the Memcheck tab and then 'Run Check' the program argument is not passed along. Thus my program exits straight away, as it should. This, however, makes running Memcheck from within Codelite fruitless and I have to resort to running it manually from a terminal.

Did I miss some setting about passing on program arguments? Is this intended behaviour? Or just a bug?

I am using the IDE on Ubuntu 14.0.4 and I am running the latest version found in the https://repos.codelite.org/ubuntu/ trusty repository: v9.1.1.

**Comment by AJenbo** [comment]
Thank you for the report, I will try and look in to the issue some time next week.

**Comment by eranif** [comment]
I just tested this on Linux - this works properly (i.e. the program arguments were passed to valgrind)
Make sure that the program arguments are set under the field:
`Project settings->General->Execution section->Program Arguments` and NOT under the "Debuggig" section

**Comment by Nathilion** [comment]
Ah, I figured out where it went wrong. I did have my program argument in the execution section but it wasn't passed on properly. My arguments looks like this: `{\"Ordinal\":7}`. It's a piece of JSON we use to discriminate multiple instances of our program. With the regular debugger the \ escapes the quote character and the argument is passed on properly as the string `{"Ordinal":7}`.

With valgrind this results in `{Ordinal:7}` being passed on. Note how the quotation marks are missing and thus the JSON parser did not recognize it properly. I figured I'd try the single quote ticks to escape my double quotes and that did the trick of passing it on properly. The argument is now: `{'"'Ordinal'"':7}`.

It seems to me that two different argument parsers are used to assemble the command line. Is there a way this could be changed so that there's one only type of parser for all the possible debuggers and plugins?
